TwitchLib.Api/Extensions/System/IEnumerableExtensions.cs

```diff
@@ -0,0 +1,12 @@
+﻿using System.Collections.Generic;
+
+namespace TwitchLib.Api.Extensions.System
+{
+    public static class IEnumerableExtensions
+    {
+        public static void AddTo<T>(this IEnumerable<T> source, List<T> destination)
+        {
+            if (source != null) destination.AddRange(source);
```

Comment:
I agree. A new ArgumentNullException(nameof(source)) should be thrown when the source is null.
At first I thought it wasn't needed because it's an extension method, but I forgot you can statically call these methods as well.
Secondly, I'd also add a check for the destination parameter.
Since this is a public extension, a summary would be nice as well

Reply:
It should be internal to the library and not externally usable.
Throw ArgumentNullException on both source and destination, if it's intended to have this extension used elsewhere.

Doesn't change your code in any way as you are already implementing Elvis (safe navigation) operators on the resultset before calling AddTo, so in reality - in this call - source can't be null when it actually hits this code.
What I would do is remove the Elvis operators after both the where predicates in the call as they aren't needed - if you already are up to that point in navigation you are fine - it'll just add 0 items when AddRange is called.
Sorry about those extras - wrote it quickly in the middle of a trash fight in the middle of Antorus Mythic :D

Reply:
^ Agreed the full 100%. Noticed the null checks as well after Where() methods, which would never return null. So unneeded.